src/components/organisms/ChannelInfoForm/ChannelInfoForm.tsx

```diff
@@ -0,0 +1,174 @@
+    } else {
+      setValidCheckPassword(false);
+      setHelperTextCheckPassword('비밀번호 일치하지 않음');
+    }
+    setCheckPassword(value);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    // FIXME: API 확정된 후 추가하기
+  };
+
+  return (
+    <Grid container justifyContent="center">
+      <Grid
+        item
+        container
+        className={classes.root}
+        direction="column"
+        justifyContent="space-evenly"
+        spacing={3}
+      >
+        <Typo variant="h3" align="center" gutterBottom>Channel Form</Typo>
+        <Typo className={classes.margin}>* 표시: 필수 입력 항목</Typo>
+        <form onSubmit={handleSubmit}>
+          <Grid item container alignItems="center" justifyContent="center">
+            <Input
+              onChange={handleChannelNameChange}
+              label="채널명 입력 *"
+              value={channelName}
+              helperText={helperTextChannelName}
+              error={!isValidChannelName}
+            />
+          </Grid>
+          <Grid item container alignItems="center" justifyContent="center">
+            <Switch
+              name="private toggle"
+              checked={isToggleChecked}
+              onChange={() => { setToggleCheck(!isToggleChecked); }}
+            />
+            <Typo>Private</Typo>
+          </Grid>
+          <Grid>
+            {isToggleChecked && (
+              <Grid
+                item
+                container
+                direction="column"
+                justifyContent="center"
+                alignItems="center"
+              >
+                <Grid item>
+                  <Input
+                    onChange={handlePasswordChange}
+                    label="비밀번호 입력 *"
+                    type="password"
+                    value={password}
+                    helperText={helperTextPassword}
+                    error={!isValidPassword}
+                    autoComplete
+                  />
+                </Grid>
+                <Grid item>
+                  <Input
+                    onChange={handleCheckPasswordChange}
+                    label="비밀번호 검증 입력 *"
+                    type="password"
+                    value={checkPassword}
+                    helperText={helperTextCheckPassword}
+                    error={!isValidCheckPassword}
+                    autoComplete
+                  />
+                </Grid>
+              </Grid>
+            )}
+          </Grid>
+          <Grid container justifyContent="center">
+            <Button
+              variant="text"
+              className={classes.button}
+              onClick={() => { setOpen(false); }}
+            >
+              Cancel
+            </Button>
+            <Button
+              type="submit"
+              disabled={(!isToggleChecked && !isValidChannelName)
+                || (isToggleChecked && (!isValidPassword || !isValidCheckPassword
+                  || !isValidChannelName))}
```

Comment:
이 코멘트는 우선 본론으로 들어가기 전에! 코멘트 내용이 취향 차이의 영역인 것 같아서 수정 요청이라기 보다는 아이디어 제시 정도로 봐주시면 좋을 것 같아요. 😄
코드를 읽을때 조건문이 이렇게 구성되어 있으면 어떤 조건에서 disable 되는지 판단하기 어려울 것 같은데, 가독성을 위해 [이런 식의 짧은 함수](https://github.com/404-DriverNotFound/frontend-b/blob/553c1e8490e5c46c3c890f1bc71134907e5a4132/src/components/organisms/UserInfoForm/UserInfoForm.tsx#L204)로 빼서 [적용해보는 건](https://github.com/404-DriverNotFound/frontend-b/blob/553c1e8490e5c46c3c890f1bc71134907e5a4132/src/components/organisms/UserInfoForm/UserInfoForm.tsx#L258) 어떨까요? 재사용되지는 않는 부분이라 굳이 싶을 수도 있긴 한데 저는 이런 부분 정리하는 걸 좋아해서 ㅎㅎ 한번 말씀드려봅니다!

Reply:
동의합니다. 그렇게 수정하는 방법이 더 좋은 것같습니다! 
저도 이 코드를 작성하다가 가독성이 떨어진 것 같다고 생각했고, 이 부분을 어떻게 해야 더 좋은 코드로 만들 지 결론을 못 냈었습니다. 그런데, 알려주신 방법대로 하면 더 좋은 코드가 될 것같습니다. 감사합니다!!

Reply:
해당 코드 이렇게 수정했습니다!! 감사합니다!
```suggestion
  const isValidForm = () => {
    if (!isToggleChecked) {
      if (isValidChannelName) return true;
      return false;
    }
    if (isValidChannelName && isValidPassword && isValidCheckPassword) return true;
    return false;
  };
```